pkg/metrics/config.go

```diff
@@ -1,11 +1,23 @@
 package metrics
 
+import (
+	"github.com/pkg/errors"
+)
+
 // PrometheusConfig defines Prometheus configuration.
 type PrometheusConfig struct {
-	Url string `yaml:"url"`
+	Url      string `yaml:"url"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
 }
 
 // Validate checks constraints in the supplied Prometheus configuration and returns an error if they are violated.
 func (c *PrometheusConfig) Validate() error {
+	if c.Url == "" && (c.Username != "" || c.Password != "") {
+		return errors.New("credentials cannot be provided without a URL")
```

Comment:
I would relax this. If no URL is specified, the feature is disabled. I don't think we should require all other configurations to be removed as well.